Documentation/src/case-studies/calendar.md

```diff
@@ -0,0 +1,350 @@
+    => ImmutableList<string>.Empty
+        .Add(CenteredMonthName(month))
+        .AddRange(FormatWeeks(month))
+        .Add($"{string.Empty,WidthOfAWeek}");
+```
+
+Now we dive into the helper functions. First we take a look at the name of the month. The only noteworthy detail is the very functional mindest seen in the solution to the centering problem. It uses a pattern match to fill in the missing spaces: it is not very efficent, but easy to understand. The recursion will be very short because our lines are only 21 characters wide.
+
+
+```cs
+private static string CenteredMonthName(IEnumerable<DateOnly> month)
+    => month
+        .First()
+        .ToString(MonthNameFormat)
+        .Center(WidthOfAWeek );
+
+
+internal static class StringExtensions
+{
+    public static string Center(this string text, int width)
+        => (width - text.Length) switch
+        {
+            0 => text,
+            1 => $" {text}",
+            _ => Center($" {text} ", width),
+        };
+}
+```
+
+We have already seen the heart of FormatWeeks in the yield solution, but now it is a separate function. FormatWeeks again needs 2 very simple helper functions, the first one Projects the Week of the year, the other one will format a sequence of days.
+
+The sequence of days can be either a complete week, or a partial week from the beginning or the end of the month. But because of the way we construct these sequences, there always is at least one element in it. 
+
+```cs
+private static IEnumerable<string> FormatWeeks(IEnumerable<DateOnly> month)
+            => month
+                .AdjacentGroupBy(GetWeekOfYear)
+                .Select(FormatWeek);
+```
+
+The GetWeekOfYear function is just calling the API with the correct parameters and always using the current culture and a little fiddling with DateTime (because DateOnly was introduced far too late in C# 10).
+
+```cs
+private static int GetWeekOfYear(DateOnly dateTime)
+            => CultureInfo
+                .CurrentCulture
+                .Calendar
+                .GetWeekOfYear(dateTime.ToDateTime(default), CultureInfo.CurrentCulture.DateTimeFormat.CalendarWeekRule, CultureInfo.CurrentCulture.DateTimeFormat.FirstDayOfWeek);
+```
+
+We are almost done with `FormatMonth`, now we really format the week, each day has a width of 3 characters, we see that in FormatDay. This means each line should have the same width of 21 characters. We simply pad the week correctly with the PadWeek function.
+
+```cs
+private static string FormatWeek(IGrouping<int, DateOnly> week)
+  => PadWeek(week.Select(FormatDay).ConcatToString(), week);
+  
+private static string FormatDay(DateOnly day)
+  => $"{day.Day,WidthOfDay}";
+```
+
+We can ignore the full weeks, because they are already 21 characters long. How do we distinguish the beginning of the month from the end? The week at the end of the month must start with the first day of the week. So wee pad accordingly from left or the right.
```

Comment:
either you say "from the left or the right" or you say "from the left or right side" or "from left or right"

Reply:
Fixed in 7a959c17f0e038b091d3d95576bd1479a2d07519